### PR TITLE
Create employee components

### LINKED ISSUE #49: Add or Remove Employees
Create a page that allows the user to add or remove employees from the employee table as mentioned in #34

Reliant on #34

### PR DESCRIPTION
**Pull request by peterle08**
Closes #49

### CODE REVIEW

File: src/app/edit-employee/edit-employee.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (kyletpugh) [inline src/app/edit-employee/edit-employee.component.spec.ts]: These .spec.ts files are unnecessary, they are for testing using jasmine and we are using cucumber.

File: src/app/edit-employee/edit-employee.component.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (kyletpugh) [inline src/app/edit-employee/edit-employee.component.html:1]: This should be basically the same as edit employee but already populated with a remove/delete option, you could have probably reused the same component with query parameters.

File: src/app/insert-employee/insert-employee.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (kyletpugh) [inline src/app/insert-employee/insert-employee.component.spec.ts]: Unnecessary file for the same reason as the other .spec.ts files.

File: src/app/list-employee/list-employee.component.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
+
```
Comment (kyletpugh) [inline src/app/list-employee/list-employee.component.html:1]: Why is the list employees not the page that lists all the current employees and has an option to add a new employee? This should be the add employee content.

File: src/app/list-employee/list-employee.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (kyletpugh) [inline src/app/list-employee/list-employee.component.spec.ts]: Unnecessary file for the same reason as all the other .spec.ts files.

File: src/app/app.module.ts [hunk 6]
```diff
@@ -9,12 +9,18 @@ import { LandingPageComponent } from './landing-page/landing-page.component';
...
+import { InsertEmployeeComponent } from './insert-employee/insert-employee.component';
+import { ListEmployeeComponent } from './list-employee/list-employee.component';
+import { EditEmployeeComponent } from './edit-employee/edit-employee.component';
 
 const appRoutes: Routes = [
   { path:'landing', component:LandingPageComponent },
   { path:'login' , component:LoginPageComponent },
   { path:'registration', component:RegisterPageComponent },
   { path:'about' , component:AboutPageComponent },
+  { path:'insert_employee', component:InsertEmployeeComponent },
```
Comment (kyletpugh) [inline src/app/app.module.ts]: This should be `path: '/employees/add'`

File: src/app/app.module.ts [hunk 7]
```diff
@@ -9,12 +9,18 @@ import { LandingPageComponent } from './landing-page/landing-page.component';
...
+  { path:'list_employee', component:ListEmployeeComponent },
```
Comment (kyletpugh) [inline src/app/app.module.ts]: This should be `path: '/employees'`

File: src/app/app.module.ts [hunk 8]
```diff
@@ -9,12 +9,18 @@ import { LandingPageComponent } from './landing-page/landing-page.component';
...
+  { path:'edit_employee', component:EditEmployeeComponent},
```
Comment (kyletpugh) [inline src/app/app.module.ts]: This should be `path: '/employees/edit'`

**Review by kyletpugh — CHANGES_REQUESTED** [review]
Where is the list of employees? The employees does not belong in the header, and should link to employees list not adding an employee. The edit employee does not correctly navigate and is not correct, you need to be able to edit the same attributes as when they were added, but also be able to remove the user, why enter the employee id twice to do so? Just make a remove/delete button and popup a confirmation. You need to fix these things before the pull request is approved.

File: src/app/insert-employee/insert-employee.component.html [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+          <div><label>First name</label><input type="text" placeholder="First name" name="first_name" required></div>
+          <div><label>Last name:</label><input type="text" placeholder="Last name" name="last_name" required></div>
+          <div><label>Middle name:</label><input type="text" placeholder="Middle name" name="middle_name" required></div>
+          <div><label>SSN:</label><input type="text" name="SSN" placeholder="SSN" required></div>
+          <div>
+              <label>Gender:</label><select style="font-size:20px" name="gender" required>
+                <option selected hidden value="">select</option>
+                <option value = "M">Male</option>
+                <option value = "F">Female</option>
+                <option value = "O">Others</option>
```
Comment (kyletpugh) [inline src/app/insert-employee/insert-employee.component.html:14]: You may want to add a custom option instead that allows the person to enter their own gender, or find a comprehensive list.

**Review by kyletpugh — APPROVED** [review]

**Review by RyanYoak — APPROVED** [review]

**Pull request merged**